plugins/scaffolder/src/next/OngoingTask/OngoingTask.tsx

```diff
@@ -59,6 +61,7 @@ export const OngoingTask = (props: {
   );
 
   const [logsVisible, setLogVisibleState] = useState(false);
+  const [clickedToCancel, setClickedToCancel] = useState<boolean>(false);
```

Comment:
I can move it, why i place it here is to make the calculation of "cancelability" at one place.

```
const cancelEnabled = !(
    taskStream.cancelled ||
    clickedToCancel ||
    taskStream.completed
  );
```

If we will move this line into `ContextMenu`, it will make to read the code a bit harder, is it option disabled now or not...

Reply:
I tried to move it, and then we need pass `taskStream` inside ContextMenu

<img width="1410" alt="Screenshot 2023-03-08 at 21 24 59" src="https://user-images.githubusercontent.com/2265094/223841792-ed730778-c0da-483e-b30c-75dfaedc69b2.png">

I can move it inside, but it will add unnecessary logic to ContextMenu. 
Up to you.

Reply:
I think you can move the `useCallback` and and the state inside the button, and also be able to drive the state from a prop too, something like `disabled` or something?